pdns/dnsdistdist/libssl.cc

```diff
@@ -72,7 +72,11 @@ static int s_keyLogIndex{-1};
 void registerOpenSSLUser()
 {
   if (s_users.fetch_add(1) == 0) {
-#if (OPENSSL_VERSION_NUMBER < 0x1010000fL || defined LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER >= 0x1010000fL && !defined LIBRESSL_VERSION_NUMBER)
```

Comment:
`OPENSSL_init_crypto` has been available in libressl for a few years already...

Reply:
specifically the test can be:
`if (OPENSSL_VERSION_NUMBER >= 0x1010000fL && (!defined LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER >= 0x2070000fL)`

Reply:
Does this version of LibreSSL really support automatic initialization and, more importantly, automatic setup of thread-specific callbacks? Otherwise this will result in random memory corruption..
I really wish LibreSSL had not fiddled with `OPENSSL_VERSION_NUMBER`...

Reply:
`CRYPTO_set_id_callback` and `CRYPTO_set_locking_callback` are not needed in libressl anymore since `LIBRESSL_VERSION_NUMBER >= 0x2090100fL`. So to be conservative, we could check for this value.

Reply:
Thanks! I pushed a commit updating these version number checks, I'd be great if you could check I did not mess anything up :)